src/Covenant/Analysis/Poetry/PoetryAnalyzer.cs

```diff
@@ -0,0 +1,207 @@
+namespace Covenant.Analysis.Poetry;
+
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+using Tomlyn.Model;
+
+internal class PoetryAnalyzer(IFileSystem fileSystem, IEnvironment environment) : Analyzer
+{
+    private const string ExcludeGroups = "--exclude-poetry-groups";
+    private const string DisablePoetry = "--disable-poetry";
+    private const string VirtualEnvironmentPath = "--virtual-environment-path";
+
+    private readonly PoetryAssetReader _assetReader = new PoetryAssetReader(fileSystem, environment);
+    private readonly IEnvironment _environment = environment;
+    private bool _enabled = true;
+    private DirectoryPath? _virtualEnvironmentPath;
+
+    public override bool Enabled => _enabled;
+    public override string[] Patterns { get; } = ["**/pyproject.toml"];
+
+    public override void Analyze(AnalysisContext context, FilePath path)
+    {
+        path = path.GetDirectory().CombineWithFilePath("pyproject.toml");
+
+        // Read the asset file
+        var assetFile = _assetReader.ReadAssetFile(path);
+        if (assetFile == null)
+        {
+            context.AddError("Could not read pyproject.toml");
+            return;
+        }
+
+        // Add main component
+        var root = context.AddComponent(
+            new PoetryComponent(assetFile.Tool.Poetry.Name!, assetFile.Tool.Poetry.Version!, BomComponentKind.Root));
```

Comment:
The build errors that GitHub is showing me are occurring here (on my screen, it's showing me a bunch of GitHub Actions failures with `Dereference of a possibly null reference`) are exactly the kind of problem I'd expect when you try to silence compiler warnings about nullability on data serialization types by slapping a `?` on everything. It just pushes the warnings elsewhere in the code.